sql/src/test/java/org/apache/druid/sql/calcite/CalciteTableAppendTest.java

```diff
@@ -0,0 +1,214 @@
+                new Object[] {"def", null, null, null},
+                new Object[] {"abc", null, null, null},
+                new Object[] {"", "a", 1.0D, 1.0F},
+                new Object[] {"10.1", "a", 1.7D, 0.1F},
+                new Object[] {"2", "a", 0.0D, 0.0F},
+                new Object[] {"1", "b", null, null},
+                new Object[] {"def", "b", null, null},
+                new Object[] {"abc", "b", null, null}
+            )
+        )
+        .expectedLogicalPlan(
+            ""
+                + "LogicalProject(exprs=[[$1, $8, $11, $13]])\n"
+                + "  LogicalTableScan(table=[[APPEND]])\n"
+        )
+        .run();
+  }
+
+  @Test
+  public void testAppendSameTableMultipleTimes()
+  {
+    testBuilder()
+        .sql("select dim1,dim4,d1,f1 from TABLE(APPEND('foo','numfoo','foo')) u where dim1='2'")
+        .expectedResults(
+            ResultMatchMode.RELAX_NULLS,
+            ImmutableList.of(
+                new Object[] {"2", null, null, null},
+                new Object[] {"2", null, null, null},
+                new Object[] {"2", "a", 0.0D, 0.0F}
+            )
+        )
+        .run();
+  }
+
+  @Test
+  public void testAppendtSingleTableIsValid()
+  {
+    testBuilder()
+        .sql("select dim1 from TABLE(APPEND('foo')) u")
+        .expectedResults(
+            ImmutableList.of(
+                new Object[] {""},
+                new Object[] {"10.1"},
+                new Object[] {"2"},
+                new Object[] {"1"},
+                new Object[] {"def"},
+                new Object[] {"abc"}
+            )
+        )
+        .run();
+  }
+
+  @Test
+  public void testAppendtNoTableIsInvalid()
+  {
+    try {
+      testBuilder()
+          .sql("select dim1 from TABLE(APPEND()) u")
+          .run();
+      Assert.fail("query execution should fail");
+    }
+    catch (DruidException e) {
+      MatcherAssert.assertThat(
+          e,
+          invalidSqlIs("No match found for function signature APPEND() (line [1], column [24])")
+      );
+    }
+  }
+
+  @Test
+  public void testAppendNoTableIsInvalid()
+  {
+    try {
+      testBuilder()
+          .sql("select dim1 from TABLE(APPEND()) u")
+          .run();
+      Assert.fail("query execution should fail");
+    }
+    catch (DruidException e) {
+      MatcherAssert.assertThat(
+          e,
+          invalidSqlIs("No match found for function signature APPEND() (line [1], column [24])")
+      );
+    }
+  }
+
+  @Test
+  public void testAppendtSingleTableIsInvalidArg()
+  {
+    try {
+      testBuilder()
+          .sql("select dim1 from TABLE(APPEND('foo',111)) u")
+          .run();
+      Assert.fail("query execution should fail");
+    }
+    catch (DruidException e) {
+      MatcherAssert.assertThat(
+          e,
+          invalidSqlIs(
+              "All arguments to APPEND should be literal strings. Argument #1 is not string (line [1], column [37])"
```

Comment:
do you think we should make the argument index printing to be 1 based instead of 0 based? 
Also, I'm not sure whether it is possible to print the argument itself too, if we can do that then that'd be great too.